### PR TITLE
Adds support for missing Olimex boards

### PR DESCRIPTION
**Pull request by Stanimir-Petev**
Olimex employee here. We were missing few boards from the list and customers requested we add them. Boards added are:

ESP32-S2-DevKit-Lipo
ESP32-S2-DevKit-Lipo-USB
ESP32-S3-DevKit-Lipo
ESP32-C3-DevKit-Lipo
ESP32-C6-EVB

Also added to boards.txt and the respective directories in the "variants" folder. The boards are open-source hardware and info can be found at our website.

We tested as much as we could, but let us know if we missed something.

Thanks espressif and Arduino team.

### CODE REVIEW

File: variants/esp32c3-devkit-lipo/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+static const uint8_t RX = 20;
+
+// define I2C pins 
+//static const uint8_t SDA = 8;
+//static const uint8_t SCL = 9;
+// define SPI pins
+//static const uint8_t SS    = 7;
+//static const uint8_t MOSI  = 6;
+//static const uint8_t MISO  = 5;
+//static const uint8_t SCK   = 4;
```
Comment (P-R-O-C-H-Y) [inline variants/esp32c3-devkit-lipo/pins_arduino.h:25]: Those pins needs to be declared.

File: variants/esp32c6-evb/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	static const uint8_t TX1 = 5;
+	static const uint8_t RX1 = 4;
+
+	static const uint8_t SDA = 6;
+	static const uint8_t SCL = 7;
+
+	static const uint8_t SS    = 21;
+	static const uint8_t MOSI  = 18;
+	static const uint8_t MISO  = 20;
+	static const uint8_t SCK   = 19;
```
Comment (P-R-O-C-H-Y) [inline variants/esp32c6-evb/pins_arduino.h:35]: Formatting.
```suggestion
static const uint8_t TX1 = 5;
static const uint8_t RX1 = 4;

static const uint8_t SDA = 6;
static const uint8_t SCL = 7;

static const uint8_t SS    = 21;
static const uint8_t MOSI  = 18;
static const uint8_t MISO  = 20;
static const uint8_t SCK   = 19;
```

File: variants/esp32s2-devkit-lipo-usb/pins_arduino.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+// define I2C pins 
+//static const uint8_t SDA = 8;
+//static const uint8_t SCL = 9;
+
+// define SPI pins
+//static const uint8_t SS    = 34;
+//static const uint8_t MOSI  = 35;
+//static const uint8_t MISO  = 37;
+//static const uint8_t SCK   = 36;
```
Comment (P-R-O-C-H-Y) [inline variants/esp32s2-devkit-lipo-usb/pins_arduino.h:26]: I2C and SPI pins needs to be declared.

File: variants/esp32s2-devkit-lipo/pins_arduino.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+// define I2C pins 
+//static const uint8_t SDA = 8;
+//static const uint8_t SCL = 9;
+
+// define SPI pins
+//static const uint8_t SS    = 34;
+//static const uint8_t MOSI  = 35;
+//static const uint8_t MISO  = 37;
+//static const uint8_t SCK   = 36;
```
Comment (P-R-O-C-H-Y) [inline variants/esp32s2-devkit-lipo/pins_arduino.h:26]: I2C and SPI pins needs to be declared.

File: variants/esp32s3-devkit-lipo/pins_arduino.h [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+	// I2C
+	static const uint8_t SDA = 48;
+	static const uint8_t SCL = 47;
+
+	// SPI
+	static const uint8_t SS    = 10;
+	static const uint8_t MOSI  = 11;
+	static const uint8_t MISO  = 13;
+	static const uint8_t SCK   = 12;
```
Comment (P-R-O-C-H-Y) [inline variants/esp32s3-devkit-lipo/pins_arduino.h:34]: Formatting.
```suggestion
// UART
static const uint8_t TX1 = 17;
static const uint8_t RX1 = 18;

// I2C
static const uint8_t SDA = 48;
static const uint8_t SCL = 47;

// SPI
static const uint8_t SS    = 10;
static const uint8_t MOSI  = 11;
static const uint8_t MISO  = 13;
static const uint8_t SCK   = 12;
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@Stanimir-Petev PTAL on my comments.

File: variants/esp32s3-devkit-lipo/pins_arduino.h (hunk 5 above)
Comment (Stanimir-Petev) [reply variants/esp32s3-devkit-lipo/pins_arduino.h]: Thank you for the suggestions, I made the changes.
Let me know if there are other issues.

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**